main.tf

```diff
@@ -93,11 +85,8 @@ module "emr-cluster" {
   custom_ami_id                                     = var.custom_ami_id
 
   # Security groups
-  emr_managed_master_sg_id    = module.emr-sgs.emr_managed_master_sg_id
-  emr_additional_master_sg_id = module.emr-sgs.emr_additional_master_sg_id
-  emr_managed_core_sg_id      = module.emr-sgs.emr_managed_core_sg_id
-  emr_additional_core_sg_id   = module.emr-sgs.emr_additional_core_sg_id
-  emr_service_access_sg_id    = module.emr-sgs.emr_service_access_sg_id
+  emr_managed_master_sg_ids = var.emr_managed_master_sg_ids
+  emr_managed_core_sg_ids   = var.emr_managed_core_sg_ids
```

Comment:
You're only passing through the user defined values.

You'll want to also pass in the output of module `emr-sgs` so that they get applied to the cluster.

My guess is whatever you're using in the examples for passed in SGs is what's allowing your testing to work, otherwise I'm not sure why it's working.